.github/workflows/shinyapps_deploy.yml

```diff
@@ -0,0 +1,99 @@
+# Workflow derived from https://github.com/r-lib/actions/tree/master/examples
+# Need help debugging build failures? Start at https://github.com/r-lib/actions#where-to-find-help
+
+name: shiny-deploy
+
+on:
+  push:
+    branches:
+      - shiny-server-main
+      - switch-python-venv
+      - poetry
+      - release*
+
+jobs:
+  shiny-deploy:
+    runs-on: ubuntu-latest
+    # This image seems to be based on rocker/r-ver which in turn is based on debian
+    container: rocker/rstudio
+    env:
+      CONDA_ENV_NAME: virtual_env
+      # This should not be necessary for installing from public repo's however remotes::install_github() fails without it.
+      GITHUB_PAT: ${{ secrets.REPO_PAT }}
+
+    steps:
+      - name: Install System Dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y pip python3.8-venv libcurl4-openssl-dev
+ 
+      - uses: actions/checkout@v2
+
+      - uses: r-lib/actions/setup-pandoc@v1
+
+      - name: Create and Activate Python Virtual Environment
+        shell: bash
+        run: |
+          python3 -m venv "${{ env.CONDA_ENV_NAME }}"
+          chmod 755 "${{ env.CONDA_ENV_NAME }}"/bin/activate
+          source "${{ env.CONDA_ENV_NAME }}"/bin/activate
+        
+      - name: Install Python Dependencies
+        shell: bash
+        run: |
+          source "${{ env.CONDA_ENV_NAME }}"/bin/activate
+          # install 'poetry' and then use it to install 'schematic'
+          pip3 install poetry
+          git clone --single-branch --branch develop https://github.com/Sage-Bionetworks/schematic.git
+          cd schematic
+          poetry build
+          pip3 install dist/schematicpy-1.0.0-py3-none-any.whl
+          # This was the old way: pip3 install -r requirements.txt
+          # However, per Anthony Williams, "poetry will install all of the dependencies necessary for the app to run"
+          #
+          # This is to fix the error we otherwise see when the app starts up:
+          # No such file or directory: 'tests/data/example.model.jsonld'
+          mkdir -p ../tests/data/
+          cp -R tests/data/ ../tests/
+        
+      - name: Install R packages
+        run: |
+          R -f install-pkgs.R
+          
+      - name: zip virtual env
+        # ShinyApps has a limit of 7000 files, far exceeded by the many Python dependencies
```

Comment:
@brucehoff thanks for pointing out this workaround. The dependencies in python quickly add up, but point taken. We will 1) in general, ensure only the dependencies we need are part of schematic's setup; and 2) switch code organization using python subpackages to be better able to control schematic's setup. In the context of the DC app specifically, we will transition away from local installation of schematic altogether, and switch to endpoints exposing schematic's functionality via REST API.